### PR TITLE
Add support for readonly members

### PR DESCRIPTION
**Pull request by SergeyTeplyakov**

### CODE REVIEW

**Comment by SergeyTeplyakov** [comment]
Fixes #186

File: src/ErrorProne.NET.StructAnalyzers.Tests/MakeStructMemberReadOnlyCodeFixProviderTests.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+using NUnit.Framework;
+using System.Threading.Tasks;
+using VerifyCS = ErrorProne.NET.TestHelpers.CSharpCodeFixVerifier<
+    ErrorProne.NET.StructAnalyzers.MakeStructMemberReadOnlyAnalyzer,
+    ErrorProne.NET.StructAnalyzers.MakeStructMemberReadOnlyCodeFixProvider>;
+
+namespace ErrorProne.NET.StructAnalyzers.Tests
+{
+    [TestFixture]
+    public class MakeStructMemberReadOnlyCodeFixProviderTests
```
Comment (sharwell) [inline src/ErrorProne.NET.StructAnalyzers.Tests/MakeStructMemberReadOnlyCodeFixProviderTests.cs:11]: 💡 Code fix tests shouldn't be separated from analyzer tests. Separating them means you're either losing test coverage or duplicating work.

File: src/ErrorProne.NET.StructAnalyzers/ReadOnlyAnalyzer.cs [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            var syntax = symbol.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax();
+            if (syntax == null)
+            {
+                return false;
+            }
+
+            // Unfortunately, the approach solely based on dataflow analysis doesn't work
+            // because its impossible to separate the following cases:
+            // ref readonly S s = ref this; // fine
+            // ref S s = ref this; // not fine
```
Comment (sharwell) [inline src/ErrorProne.NET.StructAnalyzers/ReadOnlyAnalyzer.cs:114]: /cc @mavasani

**Comment by SergeyTeplyakov** [comment]
> 💭 We should really be providing this analyzer either as part of the Roslyn code style package, or an part of dotnet/roslyn-analyzers

Yep. I agree. One that makes the struct readonly and this one is the entire sturct can't be readonly.
I can try to move the analyzer into another place if you we can agree what's the right place for them:)

File: src/ErrorProne.NET.StructAnalyzers.Tests/MakeStructMemberReadOnlyCodeFixProviderTests.cs (hunk 1 above)
Comment (SergeyTeplyakov) [reply src/ErrorProne.NET.StructAnalyzers.Tests/MakeStructMemberReadOnlyCodeFixProviderTests.cs:11]: Even though I see the point of keeping these tests together, I actually think it makes sense to keep them separate.

My logic is: I tend to cover the corner cases for the analyzers in the analyzer's tests and have a few tests for the fixer to cover only the cases related to code transformation.

I tend to have significantly more code in the analyzers and the fixers usually significantly simpler.
